napi.h

```diff
@@ -312,30 +313,31 @@ using MaybeOrValue = T;
 ///
 /// In the V8 JavaScript engine, a Node-API environment approximately
 /// corresponds to an Isolate.
-class Env {
- private:
-  napi_env _env;
+class NogcEnv {
+ protected:
```

Comment:
Should we make this `private` and have `Env` be a friend class? With it being `protected`, the default finalizers (as well as `_env`) would be exposed in any user-defined classes that extend `NogcEnv`.